### PR TITLE
feat: check asar existence and integrity

### PR DESCRIPTION
**Pull request by demetris-manikas**

### CODE REVIEW

**Comment by demetris-manikas** [comment]
Forgot to mention that this patch was created in response to #371. 
Doesn't solve the problem but throws more descriptive errors.

**Comment by develar** [comment]
If we want to report more clear message if asar file doesn't exists, maybe we should fix code 

```
try {
        return statFile(path.join(resourcesDir, "app.asar"), relativeFile) != null
      }
      catch (e) {
        // asar throws error on access to undefined object (info.link)
        return false
      }
```

? Here we just ignore error and returns false. I think, it is possible to throw error `throw new Error(`Package file ${asarFile} was not created.`)` here (checking error) and avoid yet another check.

File: src/platformPackager.ts [hunk 1]
```diff
@@ -223,17 +226,38 @@ export abstract class PlatformPackager<DC extends PlatformSpecificBuildOptions>
...
+
+    if (outStat == null) {
+      throw new Error(`Package file ${asarFile} was not created.`)
+    }
+
+    try {
+      listPackage(asarFile)
+    }
+    catch (e) {
+      throw new Error(`Package file ${asarFile} is corrupted.`)
```
Comment (develar) [inline src/platformPackager.ts]: Is it real case?

Comment (demetris-manikas) [inline src/platformPackager.ts]: #371 suggests that it is real. Haven't been able to reproduce.

**Comment by demetris-manikas** [comment]
Why not. It just feels more natural to me to first check for existence and then try to act on a file than trying to act on a file and on error check for its existence.

**Comment by develar** [comment]
Makes sense since #397 I feel that there is area to optimise, but... let's do it later.

**Pull request merged**